tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -3259,4 +3264,90 @@ public function userFavoritesElementInSpaceUsingTheWebdavApi(string $user, strin
             )
         );
     }
+
+    /**
+     * returns the id of a file
+     *
+     * @param string|null $fullUrl
+     * @param string|null $user
+     * @param string|null $password
+     * @param array|null $headers
+     * @param string|null $body
+     * @return string
+     * @throws GuzzleException
+     */
+    public function getFileIdForPath(
+        ?string $fullUrl,
+        ?string $user,
+        ?string $password,
+        ?array $headers = [],
+        ?string $body
+    ): string {
+        $response= $this->sendPropfindRequestToUrl(
+            $fullUrl,
+            $user,
+            $password,
+            $this->featureContext->getStepLineRef(),
+            $headers,
+            $body
+        );
+        \preg_match(
+            '/\<oc:fileid\>([^\<]*)\<\/oc:fileid\>/',
```

Comment:
```suggestion
            '/\<oc:fileid\>([^<]+)\<\/oc:fileid\>/',
```
- :medal_military: regex looks powerful
- no escaping needed for `<` inside `[]`
- maybe the content inside `oc:fileid` tag should have at least one char. so `+` is better imo
- if only numbers are in the content then maybe using `\d+` is better but please decide.

Reply:
@kiranparajuli589 
> maybe the content inside oc:fileid tag should have at least one char. so + is better imo

`[^<]*` mean  matches any character except <  zero or more times isn't this correct fileid should not contain `<`